rticonnextdds_connector/rticonnextdds_connector.py

```diff
@@ -946,23 +948,40 @@ def __init__(self, connector, name):
 			raise ValueError("Invalid Publication::DataWriter name")
 		self.instance = Instance(self)
 
-	def write(self, options=None):
-		"""Publishes the values of the current``instance``
+	def write(self, **kwargs):
+		"""Publishes the values of the current ``instance``
 
-		Note that after writing the current ``instance``, its values remain
+		Note that after writing it, ``instance``'s values remain
 		unchanged. If for the next write you need to start from scratch, use
 		:meth:`clear_members()`
 
+		This method receives a number of optional parameters, a subset of those
+		documented in the `Writing Data section <https://community.rti.com/static/documentation/connext-dds/current/doc/manuals/connext_dds/html_files/RTI_ConnextDDS_CoreLibraries_UsersManual/index.htm#UsersManual/Writing_Data.htm?Highlight=DDS_WriteParams_t>`__. 
+		of the *Connext DDS Core Libraries** User's Manual.
+
+		The supported parameters are:
+
+		:param integer source_timestamp: The source timestamp, an integer representing the total number of nanoseconds
+		:param dict identity: A dictionary containing the keys ``"writer_guid"`` (a list of 16 bytes) and ``"sequence_number"`` (an integer) that uniquely identifies this sample.
+		:param dict related_sample_identity: Used for request-reply communications. It has the same format as ``identity``
+
+		For example::
+
+			output.write(
+				identity={"writer_guid":[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15], "sequence_number":1},
```

Comment:
Should the GUID not be 16-bytes in length?

Reply:
I think this is an invalid GUID that would result in an error from write.

Reply:
Ok, I see in your test that in this case the GUID would be correct and you would add an aditional byte of 0 at the end. I think we should modify this so we have an actual GUID in the documentation but I'll merge it so I have the changes and do it as part of my work.